src/components/Products/Header/Header.tsx

```diff
@@ -1,17 +1,26 @@
+import { FC } from 'react';
+
+import { IProduct } from 'models/IProduct';
 import classes from './Header.module.scss';
 
-function Header() {
+interface HeaderProps {
+  products?: IProduct[];
+}
+
+const Header: FC<HeaderProps> = ({ products }) => {
   return (
     <div className={classes.header}>
       <h1>All products</h1>
       <div className={classes.header__amount}>
         <div className={classes.header__number}>
-          <span>117</span>
+          <span>{products?.length}</span>
         </div>
-        <span className={classes.header__name}>Products</span>
+        <span className={classes.header__name}>
+          {products && products?.length === 1 ? 'Product' : 'Products'}

```

Comment:
products?.length === 1 is enough. You already check for true with "?" condition.